clap_complete/src/shells/bash.rs

```diff
@@ -102,7 +102,7 @@ fn all_subcommands(cmd: &Command) -> String {
         }
     }
     let mut subcmds = vec![];
-    let fn_name = cmd.get_name().replace('-', "__");
+    let fn_name = bin_name.replace('-', "__");
```

Comment:
Since this didn't change any tests, it means we have no coverage for this case.

Could you
- Insert a commit before this one that adds a test, showing the existing "bad" behavior by passing?
- This commit should then be updated to pass, showing how behavior changed.

Reply:
I added tests to verify that the binary name is used instead of package name (without involving subcommand completions). To be consistent I added one for each shell. Since this bug only affected Bash only that test fails in the commit before the fix.